src/com/lftechnology/java/training/dipak/employeemanagement/service/EmployeeService.java

```diff
@@ -0,0 +1,165 @@
+			LOGGER.info(
+				"The username field can't be empty. Please re-check and try again.");
+			return numberOfUserInserted;
+		}
+		else if (e.getPassword().length() == 0) {
+			LOGGER.info(
+				"The password field can't be empty. Please re-check and try again.");
+			return numberOfUserInserted;
+		}
+		else if (e.getRole().equals(UserType.INVALID)) {
+			LOGGER.info(
+				"The role field should either be ADMIN or USER. Please re-check and try again.");
+			return numberOfUserInserted;
+		}
+		else if (e.getFullName().length() == 0) {
+			LOGGER.info(
+				"The fullname field can't be empty. Please re-check and try again.");
+			return numberOfUserInserted;
+		}
+
+		numberOfUserInserted = ed.addEmployee(e);
+
+		if (numberOfUserInserted == 0) {
+			LOGGER.info(
+				"Insertion failed. No users were added. Please check and enter the data again.");
+		}
+		else {
+			LOGGER.log(
+				Level.INFO, "Number of users added={0}", numberOfUserInserted);
+		}
+
+		return numberOfUserInserted;
+	}
+
+	/**
+	 * <p>This method calls the viewEmployee in DAO layer and displays the
+	 * result set obtained.</p>
+	 * 
+	 * @param e
+	 * @return
+	 */
+	@Override
+	public ResultSet viewEmployee(Employee e) {
+
+		ConsoleHandler ch = new ConsoleHandler();
+		ch.setLevel(Level.FINEST);
+		LOGGER.addHandler(ch);
+		LOGGER.setUseParentHandlers(false);
+
+		LoggerFormatter myFormat = new LoggerFormatter();
+
+		ch.setFormatter(myFormat);
+
+		ResultSet rs = null;
+
+		EmployeeDao ed = DaoFactory.getEmployeeDao();
+
+		rs = ed.viewEmployee(e);
+		try {
+			LOGGER.info(
+				"eid\t\tusername\t\tfullname\t\tdepartment\t\taddress\t\t\n");
+			while (rs.next()) {
+				LOGGER.log(
+					Level.INFO, "{0}\t\t{1}\t\t{2}\t\t{3}\t\t{4}\t\t\n",
+					new Object[] {
+						rs.getInt(1), rs.getString(2), rs.getString(5),
+						rs.getString(6), rs.getString(7)
+				});
+			}
+		}
+		catch (SQLException e1) {
+			LOGGER.log(Level.INFO, "Exception::{0}", e1);
+		}
+
+		return null;
+	}
+
+	/**
+	 * <p>This method calls the terminateEmployee method present in the DAO
+	 * layer and prints the number of employees terminated.</p>
+	 * 
+	 * @param e
+	 * @return numberOfTerminatedUsers
+	 */
+	@Override
+	public int terminateEmployee(Employee e) {
+
+		EmployeeDao ed = DaoFactory.getEmployeeDao();
+		int numberOfTerminatedUsers = 0;
+
+		numberOfTerminatedUsers = ed.terminateEmployee(e);
+
+		if (numberOfTerminatedUsers == 0) {
+			LOGGER.info(
+				"No users were deleted. Please check the fullname again.");
+		}
+		else {
+			LOGGER.log(
+				Level.INFO, "Number of users deleted={0}",
+				numberOfTerminatedUsers);
+		}
+
+		return numberOfTerminatedUsers;
+
+	}
+
+	/**
+	 * <p>This method calls editEmployeeDetails method present in DAO layer.</p>
+	 * 
+	 * @param e
+	 * @return e
+	 */
+	@Override
+	public Employee editEmployeeDetails(Employee employee) {
+
+		EmployeeDao ed = DaoFactory.getEmployeeDao();
+
+		employee = ed.editEmployeeDetails(employee);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "employee". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)